### PR TITLE
ar71xx: add support for LigoWawe LigoDLB Propeller 5

### PR DESCRIPTION
**Pull request by robimarko**
LigoDLB Propeller 5 is 5 GHz AP based on Qualcomm AR9342 rev 2.
https://www.ligowave.com/products/ligodlb-propeller-5
Specifications:

560/450/225/40 MHz (CPU/DDR/AHB/REF)
64MB of RAM
16 MB of FLASH (8 MB usable because of dual image)
1x 10/100 Mbps Ethernet with 24V POE IN
6x LED, 1x button
UART header on PCB
JTAG header on PCB

Currently not working:

Update from official web interface
LigoWawe uses they custom app to flash images and therefore flashing OpenWRT images does not work.

Flash instruction:

Set PC to fixed IP address 192.168.2.254
Download openwrt-ar71xx-generic-dlb-5-squashfs-sysupgrade.bin and rename it to fwupdate.bin
Start a TFTP server with the file fwupdate.bin in its root directory
Turn off the router
Press and hold Reset button
Turn on router with the reset button pressed and wait 6 or more seconds
LED-s will start flashing and after it is done device should boot into OpenWRT.

Device specifics:
The bootloader has Netconsole built it, you can access it by holding the reset button while powering on for at least 3 seconds and less than 6 seconds.
It can be useful for recovery as serial pads are on PCB and device is not meant to be opened at all.

8 MB of 16 MB of SPI flash is usable.
That is due to LigoWawe using dual firmware bootloader.
So if after 3 boots boot counter is not reset bootloader will boot the other firmware image.
Due to that at boot counter for used firmware image is reset to 0.

Signed-off-by: Robert Marko robimarko@gmail.com

### CODE REVIEW

**Comment by robimarko** [comment]
@pepe2k Any chance on merging this?
It has been open since October

File: target/linux/ar71xx/files/arch/mips/ath79/mach-ligodlb-5.c [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+				   AR934X_ETH_CFG_MII_GMAC0_SLAVE);
+
+	ath79_register_leds_gpio(-1, ARRAY_SIZE(ligdolb_leds_gpio),
+				 ligdolb_leds_gpio);
+	ath79_register_gpio_keys_polled(-1, LIGODLB_KEYS_POLL_INTERVAL,
+					 ARRAY_SIZE(ligdolb_gpio_keys),
+					 ligdolb_gpio_keys);
+	
+	ath79_register_mdio(1, 0x0);
+	ath79_register_mdio(0, 0x0);
```
Comment (pepe2k) [inline target/linux/ar71xx/files/arch/mips/ath79/mach-ligodlb-5.c]: Why do you register both mdio buses?

File: target/linux/ar71xx/files/arch/mips/ath79/machtypes.h [hunk 2]
```diff
@@ -127,6 +127,7 @@ enum ath79_mach_type {
 	ATH79_MACH_JWAP003,			/* jjPlus JWAP003 */
 	ATH79_MACH_JWAP230,			/* jjPlus JWAP230 */
 	ATH79_MACH_LIMA,			/* 8devices Lima */
+	ATH79_MACH_LIGODLB_5,			/* LigoDLB Propeller 5 */
```
Comment (pepe2k) [inline target/linux/ar71xx/files/arch/mips/ath79/machtypes.h]: Please, keep alphabetical order.

File: target/linux/ar71xx/image/generic.mk [hunk 3]
```diff
@@ -219,6 +219,15 @@ define Device/cpe870
 endef
 TARGET_DEVICES += cpe870
 
+define Device/dlb-5
+  BOARDNAME := DLB-5
+  DEVICE_PACKAGES := rssileds
+  DEVICE_TITLE := LigoDLB Propeller 5
+  MTDPARTS := spi0.0:192k(u-boot)ro,64k(u-boot-env),7680k(firmware),7680k(firmware-backup),576k(data)ro,128k(cfg)ro,64k(art)ro
+  IMAGE_SIZE := 7680k
+  endef
```
Comment (pepe2k) [inline target/linux/ar71xx/image/generic.mk]: You have a mistake here (indentation).

File: target/linux/ar71xx/mikrotik/config-default [hunk 4]
```diff
@@ -91,6 +91,7 @@
 # CONFIG_ATH79_MACH_JWAP003 is not set
 # CONFIG_ATH79_MACH_JWAP230 is not set
 # CONFIG_ATH79_MACH_LIMA is not set
+# CONFIG_ATH79_MACH_LIGODLB_5 is not set
```
Comment (pepe2k) [inline target/linux/ar71xx/mikrotik/config-default]: Please, rebase your code on current master.

File: target/linux/ar71xx/nand/config-default [hunk 5]
```diff
@@ -93,6 +93,7 @@ CONFIG_ATH79_MACH_HIVEAP_121=y
 # CONFIG_ATH79_MACH_JWAP003 is not set
 # CONFIG_ATH79_MACH_JWAP230 is not set
 # CONFIG_ATH79_MACH_LIMA is not set
+# CONFIG_ATH79_MACH_LIGODLB_5 is not set
```
Comment (pepe2k) [inline target/linux/ar71xx/nand/config-default]: Please, rebase your code on current master.

File: target/linux/ar71xx/base-files/lib/ar71xx.sh [hunk 6]
```diff
@@ -554,6 +554,9 @@ ar71xx_board_detect() {
 	*"DB120 reference board")
 		name="db120"
 		;;
+	*"LigoDLB Propeller 5")
```
Comment (pepe2k) [inline target/linux/ar71xx/base-files/lib/ar71xx.sh]: Please, keep alphabetical order.

File: target/linux/ar71xx/files/arch/mips/ath79/mach-ligodlb-5.c [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+ *  Copyright (C) 2010-2012 Gabor Juhos <juhosg@openwrt.org>
+ *  Copyright (C) 2011-2012 Anan Huang <axishero@foxmail.com>
+ *  Copyright (C) 2017 Robert Marko <robimarko@otvorenamreza.org>
+ *
+ *  This program is free software; you can redistribute it and/or modify it
+ *  under the terms of the GNU General Public License version 2 as published
+ *  by the Free Software Foundation.
+ */
+ 
+#include <linux/pci.h>
```
Comment (pepe2k) [inline target/linux/ar71xx/files/arch/mips/ath79/mach-ligodlb-5.c]: Please, drop unused/not required headers.

File: target/linux/ar71xx/files/arch/mips/ath79/mach-ligodlb-5.c [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+		.code		= KEY_RESTART,
+		.debounce_interval = LIGODLB_KEYS_DEBOUNCE_INTERVAL,
+		.gpio		= LIGODLB_GPIO_BTN_RESET,
+		.active_low	= 1,
+	}
+};
+
+static void __init ligdolb_5_setup(void)
+{
+	u8 *wifi_mac = (u8 *) KSEG1ADDR(0x1fff1002);
```
Comment (pepe2k) [inline target/linux/ar71xx/files/arch/mips/ath79/mach-ligodlb-5.c]: This is not required, that's the regular MAC offset, you can call `ath79_register_wmac(ee, NULL);`.

**Review by pepe2k — CHANGES_REQUESTED** [review]
Hi @robimarko,

Please, see my comments inline, below.

Also, please read our [submitting patches rules](https://lede-project.org/submitting-patches), make sure your pull request follows them (hints: "commit description must have less than 75 characters per line" and your SoB line is wrong) and rebase the code on current master.

Cheers,
Piotr

**Comment by robimarko** [comment]
@pepe2k Your comments have been addressed.
I apologize for obvious mistakes.
This was my first attempt at supporting devices and since then I have gotten better and a couple of a couple of my PRs adding support for other devices have been merged.

Thanks

**Comment by pepe2k** [comment]
> > @pepe2k Your comments have been addressed.

Thanks but at least my first comment about your changes in `target/linux/ar71xx/base-files/etc/board.d/01_leds ` hasn't been addressed.

Cheers,
Piotr

**Comment by pepe2k** [comment]
> Sorry,but I have no idea how to properly configure it.
> Do you have a example that has properly configured RSSI leds?

Have a look for example at N5Q or fritz300e.

Cheers,
Piotr

**Comment by robimarko** [comment]
Thanks,it should be correct now.
Also,conflict in 01_leds should be resolved

**Comment by blogic** [comment]
@robimarko sorry but  the patch lost traction and no longer applies. If you would still like to see it getting merged, please rebase push and then reopen the patch.

**Comment by robimarko** [comment]
I can rebase it but I no longer have access to the device to test out.

**Pull request closed without merging**